omnilib/prototypes/recipe-generation.lua

```diff
@@ -542,7 +533,7 @@ function ItemGen:import(item)
 		setPlace(proto.place_result):
 		setSubgroup(proto.subgroup):
 		setFuelCategory(proto.fuel_category):
-		setIcons(proto.icons or proto.icon):
+		setIcons(find_result_icon(proto)):
```

Comment:
no need to call the function here.
The function cant look for more than item.icon /icons aswell